<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1112</width>
    <height>742</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_4">
    <item>
     <widget class="QTabWidget" name="tabs">
      <property name="tabShape">
       <enum>QTabWidget::Rounded</enum>
      </property>
      <property name="currentIndex">
       <number>2</number>
      </property>
      <widget class="QWidget" name="Scientists">
       <attribute name="title">
        <string>Scientists</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QLineEdit" name="scientist_search">
          <property name="text">
           <string/>
          </property>
          <property name="alignment">
           <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
          </property>
          <property name="placeholderText">
           <string>Search..</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTableWidget" name="scientist_table">
          <property name="selectionBehavior">
           <enum>QAbstractItemView::SelectRows</enum>
          </property>
          <property name="sortingEnabled">
           <bool>true</bool>
          </property>
          <attribute name="horizontalHeaderCascadingSectionResizes">
           <bool>true</bool>
          </attribute>
          <attribute name="horizontalHeaderDefaultSectionSize">
           <number>235</number>
          </attribute>
          <attribute name="horizontalHeaderMinimumSectionSize">
           <number>100</number>
          </attribute>
          <attribute name="horizontalHeaderShowSortIndicator" stdset="0">
           <bool>true</bool>
          </attribute>
          <attribute name="horizontalHeaderStretchLastSection">
           <bool>true</bool>
          </attribute>
          <attribute name="verticalHeaderVisible">
           <bool>false</bool>
          </attribute>
          <column>
           <property name="text">
            <string>Full name</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Gender</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Year of Birth</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Year of Death</string>
           </property>
          </column>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="scientist_delete">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="autoFillBackground">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Delete scientist</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="add_label">
          <property name="layoutDirection">
           <enum>Qt::LeftToRight</enum>
          </property>
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="text">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;&lt;span style=&quot; font-size:x-large; font-weight:600;&quot;&gt;Add scientist to list&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="scientist_name">
            <property name="text">
             <string>Name:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="scientist_name_error">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLineEdit" name="scientist_name_input">
          <property name="text">
           <string/>
          </property>
          <property name="placeholderText">
           <string>Name..</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QLabel" name="scientist_gender">
            <property name="text">
             <string>Gender:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="scientist_gender_error">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLineEdit" name="scientist_gender_input">
          <property name="text">
           <string/>
          </property>
          <property name="placeholderText">
           <string>Gender..</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QLabel" name="scientist_yob">
            <property name="text">
             <string>Year of birth:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="scientist_yob_error">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLineEdit" name="scientist_yob_input">
          <property name="text">
           <string/>
          </property>
          <property name="placeholderText">
           <string>Year of birth..</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QLabel" name="scientist_yod">
            <property name="text">
             <string>Year of Death:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="scientist_yod_error">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLineEdit" name="scientist_yod_input">
          <property name="text">
           <string/>
          </property>
          <property name="placeholderText">
           <string>Year of death (leave empty if still alive)..</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="scientist_button">
          <property name="text">
           <string>Add</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="Computers">
       <attribute name="title">
        <string>Computers</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QLineEdit" name="computer_search">
          <property name="text">
           <string/>
          </property>
          <property name="alignment">
           <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
          </property>
          <property name="placeholderText">
           <string>Search..</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTableWidget" name="computer_table">
          <property name="selectionBehavior">
           <enum>QAbstractItemView::SelectRows</enum>
          </property>
          <property name="sortingEnabled">
           <bool>true</bool>
          </property>
          <attribute name="horizontalHeaderDefaultSectionSize">
           <number>265</number>
          </attribute>
          <attribute name="horizontalHeaderStretchLastSection">
           <bool>true</bool>
          </attribute>
          <attribute name="verticalHeaderVisible">
           <bool>false</bool>
          </attribute>
          <column>
           <property name="text">
            <string>Name</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Type</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Built</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Year</string>
           </property>
          </column>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="computer_delete">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Delete Computer</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;&lt;span style=&quot; font-size:x-large; font-weight:600;&quot;&gt;Add computer to list&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QLabel" name="computer_name_label">
            <property name="text">
             <string>Name:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="computer_name_error">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLineEdit" name="computer_name_input">
          <property name="placeholderText">
           <string>Name..</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <item>
           <widget class="QLabel" name="computer_type_label">
            <property name="text">
             <string>Type:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="computer_type_error">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLineEdit" name="computer_type_input">
          <property name="placeholderText">
           <string>Type..</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_7">
          <item>
           <widget class="QLabel" name="computer_built_label">
            <property name="text">
             <string>Built:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="computer_built_error">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLineEdit" name="computer_built_input">
          <property name="placeholderText">
           <string>Built (1 if built, 0 if not).. </string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_8">
          <item>
           <widget class="QLabel" name="computer_year_lable">
            <property name="text">
             <string>Year:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="computer_year_error">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLineEdit" name="computer_year_input">
          <property name="text">
           <string/>
          </property>
          <property name="placeholderText">
           <string>Year..</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="computer_button">
          <property name="text">
           <string>Add</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="linking_tab">
       <attribute name="title">
        <string>Linking</string>
       </attribute>
       <widget class="QWidget" name="horizontalLayoutWidget">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>10</y>
          <width>1071</width>
          <height>541</height>
         </rect>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_9">
         <item>
          <widget class="QTableWidget" name="scientist_link_table"/>
         </item>
         <item>
          <widget class="QTableWidget" name="computer_link_table"/>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="horizontalLayoutWidget_2">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>560</y>
          <width>1071</width>
          <height>61</height>
         </rect>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_10">
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QComboBox" name="scientist_dropdown"/>
         </item>
         <item>
          <spacer name="verticalSpacer_3">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_9">
           <item>
            <widget class="QPushButton" name="link_button">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Link</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="verticalSpacer_2">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QComboBox" name="computer_dropdown"/>
         </item>
         <item>
          <spacer name="verticalSpacer_4">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </widget>
      <widget class="QWidget" name="recycle_tab">
       <attribute name="title">
        <string>Recycle Bin</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_7">
        <item>
         <widget class="QTabWidget" name="trash_placeholder">
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="scientist_trash_tab">
           <attribute name="title">
            <string>Scientists</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <widget class="QTableWidget" name="scientist_trash_table">
              <property name="frameShape">
               <enum>QFrame::WinPanel</enum>
              </property>
              <property name="selectionBehavior">
               <enum>QAbstractItemView::SelectRows</enum>
              </property>
              <attribute name="horizontalHeaderDefaultSectionSize">
               <number>255</number>
              </attribute>
              <attribute name="horizontalHeaderStretchLastSection">
               <bool>true</bool>
              </attribute>
              <attribute name="verticalHeaderVisible">
               <bool>false</bool>
              </attribute>
              <column>
               <property name="text">
                <string>Full name</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Gender</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Year of birth</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Year of death</string>
               </property>
              </column>
             </widget>
            </item>
            <item>
             <layout class="QGridLayout" name="gridLayout_2">
              <item row="0" column="0">
               <widget class="QPushButton" name="scientist_recover_button">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="text">
                 <string>Recover selected from bin</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QPushButton" name="scientist_empty_button">
                <property name="text">
                 <string>Empty recycle bin</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="computer_trash_tab">
           <attribute name="title">
            <string>Computers</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <item>
             <widget class="QTableWidget" name="computer_trash_table">
              <property name="frameShape">
               <enum>QFrame::WinPanel</enum>
              </property>
              <attribute name="horizontalHeaderDefaultSectionSize">
               <number>255</number>
              </attribute>
              <attribute name="horizontalHeaderStretchLastSection">
               <bool>true</bool>
              </attribute>
              <attribute name="verticalHeaderVisible">
               <bool>false</bool>
              </attribute>
              <column>
               <property name="text">
                <string>Name</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Type</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Built</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Year</string>
               </property>
              </column>
             </widget>
            </item>
            <item>
             <layout class="QGridLayout" name="gridLayout_3">
              <item row="0" column="0">
               <widget class="QPushButton" name="computer_recover_button">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="text">
                 <string>Recover selected from bin</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QPushButton" name="computer_empty_button">
                <property name="text">
                 <string>Empty recycle bin</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Calendar</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_8">
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>Calendar</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <widget class="QCalendarWidget" name="calendarWidget"/>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1112</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuUltra_scientist_and_computer_database">
    <property name="title">
     <string>Ultra scientist and computer database</string>
    </property>
   </widget>
   <addaction name="menuUltra_scientist_and_computer_database"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
